### PR TITLE
Status bar, floating filter and HD Compact density styles in ag-grid theme

### LINKED ISSUE #1843: HD Compact Density for Salt AG-Grid
### Area

UI Components

### The problem

HD Compact is needed by Markets for the AG Grid Theme
Status bar is too tall for HD-compact

### The solution

Build the theme

### Alternatives and examples

The users override themselves

### Are you a JPMorgan Chase & Co. employee?

- [X] I am an employee of JPMorgan Chase & Co.

### LINKED ISSUE #2081: Salt AG-Grid theme cell content shifting on focus in high density
### Package name(s)

AG Grid Theme (@salt-ds/ag-grid-theme)

### Package version(s)

AG Grid theme @salt-ds/ag-grid-theme

### Description

Salt AG-Grid theme cell content shifting on focus in high density

### Steps to reproduce

Inspection

### Expected behavior

Fix cell content shifting on focus in high density

### Operating system

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Browser

- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

### Are you a JPMorgan Chase & Co. employee?

- [X] I am an employee of JPMorgan Chase & Co.

### LINKED ISSUE #2083: Salt AG-Grid theme Status bar styling
### Package name(s)

AG Grid Theme (@salt-ds/ag-grid-theme)

### Package version(s)

@salt-ds/ag-grid-theme

### Description

Status bar Labels font weight
Status bar height
Status bar foreground color

In general, status bar hasnt been defined in salt theme

### Steps to reproduce

inspection vs old theme

### Expected behavior

font weight needs adjusting

### Operating system

- [ ] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Browser

- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

### Are you a JPMorgan Chase & Co. employee?

- [X] I am an employee of JPMorgan Chase & Co.

### PR DESCRIPTION
**Pull request by libertymayc**
closes #1843
closes #2083
closes #2081 
cc @pedroBlazquez

### CODE REVIEW

**Comment by libertymayc** [comment]
> I noticed checkbox column is misaligned between header and body rows on the [HD Compact](https://8f3c9dac.saltdesignsystem-storybook.pages.dev/?path=/story/data-grid-ag-grid-theme--hd-compact&globals=mode:side-by-side;density:high) story. And when body row checkbox cell is focused, checkbox is shifted

will be fixed for #2081

File: packages/ag-grid-theme/css/_salt-ag-theme-mixin.scss [hunk 1]
```diff
@@ -287,6 +322,11 @@
     z-index: 1;
   }
 
+  .ag-cell {
+    height: calc(var(--salt-size-base) + var(--salt-spacing-100));
```
Comment (origami-z) [inline packages/ag-grid-theme/css/_salt-ag-theme-mixin.scss:326]: This will break multiline cells?

**Review by origami-z — APPROVED** [review]
Need to keep an eye on when to use `div[class*="ag-theme-salt"] {` v.s. `@mixin`, both seem to target multiple themes but using different approach.

Tested snapshot: https://stackblitz.com/edit/salt-ag-grid-theme-sxyd3l?file=package.json,App.jsx,index.js

**Review by lilyvc — APPROVED** [review]

**Comment by bhoppers2008** [comment]
@libertymayc is it possible to close the gap under the filter fields to the border?

**Comment by libertymayc** [comment]
> @libertymayc is it possible to close the gap under the filter fields to the border?

i'll have a look today @bhoppers2008 :)

**Pull request merged**